CommonCompilerConfig.cmake

```diff
@@ -125,6 +143,12 @@ elseif(MSVC OR MSYS OR MINGW)
   endif()
 endif()
 
+if(NOT "${PLATFORM_TYPE}" STREQUAL "arm")
+  set(src_compiler_definitions
```

Comment:
There are a couple of problems here.

1. `src_compiler_definitions` is of interest only in src/CMakeLists.txt, while lower case is reserved for variables used only within a given module.
2. This change and the corresponding change in src/CMakeLists.txt results in `-DNTA_ASM` being lost.

Pls move this logic to src/CMakeLists.txt and update it so that `-DNTA_ASM` gets appended correctly to the `src_compiler_definitions` variable that is already defined there.